R/download_d1_data.R

```diff
@@ -79,54 +83,55 @@ download_d1_data <- function(data_url, path) {
   ## Get package level metadata -----------
   if (!is.null(meta_id)) {
     message("\nDownloading metadata ", meta_id, " ...")
-    meta_obj <- dataone::getObject(d1c@mn, meta_id) 
+    meta_obj <- dataone::getObject(d1c@mn, meta_id)
     message("Download metadata complete")
     metadata_nodes <- dataone::resolve(cn, meta_id)
-    
+
     eml <- tryCatch({emld::as_emld(meta_obj, from = "xml")},  # If eml make EML object
                     error = function(e) {NULL})
-    
+
     # Get attributes ----------
     ## get entity that contains the metadata for the data object
     entities <- c("dataTable", "spatialRaster", "spatialVector", "storedProcedure", "view", "otherEntity")
     entities <- entities[entities %in% names(eml$dataset)]
-    
-    entity_objs <- purrr::map(entities, ~eml2::eml_get(eml, .x)) %>% 
+
+    entity_objs <- purrr::map(entities, ~EML::eml_get(eml, .x)) %>%
       # restructure so that all entities are at the same level
-      purrr::map_if(~!is.null(.x$entityName), list) %>% 
-      unlist(recursive = FALSE) 
-    
+      purrr::map_if(~!is.null(.x$entityName), list) %>%
+      unlist(recursive = FALSE)
+
     #sometimes url is stored in ...online$url instead of ...online$url$url
     #sometimes url needs to be decoded
-    entity_data <- entity_objs %>% 
-      purrr::keep(~any(grepl(data_id, 
+    entity_data <- entity_objs %>%
+      purrr::keep(~any(grepl(data_id,
                              purrr::map_chr(.x$physical$distribution$online$url, utils::URLdecode))))
-    
+
     if (length(entity_data) == 0) {
-      warning("No data metadata could not be found for ", data_url)
-      
+      warning("No data metadata could be found for ", data_url)
+
     } else {
-      
+
       if (length(entity_data) > 1) {
-      warning("multiple data metadata records found:\n",
+      warning("Multiple data metadata records found:\n",
               data_url,
               "\nThe first record was used")
       }
-      
+
       entity_data <- entity_data[[1]]
     }
 
-    attributeList <- suppressWarnings(eml2::get_attributes(entity_data$attributeList, eml))
-    
+    attributeList <- suppressWarnings(EML::get_attributes(entity_data$attributeList, eml))
+
     meta_tabular <- tabularize_eml(eml) %>% tidyr::spread(name, value)
-    
+
     ## Summary metadata from EML (combine with general metadata later)
     entity_meta <- suppressWarnings(list(
       Metadata_ID = meta_id[[1]],
       Metadata_URL = metadata_nodes$data$url[1],
+      Metadata_EML_Version = paste(stringr::str_extract_all(meta_tabular$eml.version, "[0-9]", simplify = TRUE), collapse = "."),
```

Comment:
```suggestion
      Metadata_EML_Version = str_extract_all(meta_tabular$eml.version, "[0-9].+[\\s]", simplify = TRUE),
```

Reply:
I went through the regex more carefully and found a simple solution that works across nodes:
```
Metadata_EML_Version = stringr::str_extract(meta_tabular$eml.version, "\\d\\.\\d\\.\\d")
```